### DISCUSSION TITLE
<title>2fa setup -- help!

### DISCUSSION THREAD

**Discussion by Curt46** [community/community#66421] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have been unable to figure out how to enable 2 factor authentication.  Using Firefox web browser with a QR code extension, I scan the QR code, enter the secret code in the space provided and click "continue" -- but nothing at all happens.

**Comment by hafiz-muhammad** [comment]
I don't know which extension you're using but I just tested out QR code 2FA for my account using a Firefox extension and it worked. Here's the extension that I used: [https://addons.mozilla.org/en-US/firefox/addon/auth-helper/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search](https://addons.mozilla.org/en-US/firefox/addon/auth-helper/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)

To add your account to a QR code scanner, follow these steps:

1. Go to the 2FA methods settings for your account where the QR code is provided.
2. Click the extension to open the scanner.
3. Click the 'Scan QR Code' icon next to the pencil edit icon.
4. Drag a square over the QR code.
5. Your account should then be added.

**Reply by Curt46** [reply]
I am told that I need some sort of other device with a special application -- I guess a mobile phone.  I don't have one, and I can't afford to get one.  Also that it is pointless to try and install a TOPT program on my Linux system since it renders the whole 2FA thing ineffective.  So I will just give up on any future contributions to GitHub.

**Reply by hafiz-muhammad** [reply]
The only 2FA program I use is on Linux. Here is what I use: https://flathub.org/apps/com.belmoussaoui.Authenticator

**Comment by yvan-solutions** [comment]
@Curt46 What do you mean by "nothing at all happens"?  Did you see a bunch of numeric codes and a download button or did the process hang at the authenticator app setup?

**Reply by Curt46** [reply]
I am told that I need some sort of other device with a special application -- I guess a mobile phone.  I don't have one, and I can't afford to get one.  Also that it is pointless to try and install a TOPT program on my Linux system since it renders the whole 2FA thing ineffective.  So I will just give up on any future contributions to GitHub.

**Comment by DNin01** [comment]
A TOTP program for your computer is **more secure than not having a place to store the code at all** (a computer is something you have - one factor, then your password is something you know - two factors), and to start, I think our priority right now should be to increase your security **at all**. If security is important to you, you can always change it later when you get the chance.

I'm assuming the reason you can't get it to work is because you don't have the right tools - a QR code scanner by itself won't work because a TOTP seed is a dynamic piece of information - it outputs a value that changes over time, typically every 30 seconds, so you will need an authenticator app. Typically, these are for phones, but there are probably some options out there for Linux.
